src/App.tsx

```diff
@@ -14,11 +15,44 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goodsList: string[],
+  isListVisible: boolean,
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    goodsList: goodsFromServer,
+    isListVisible: false,
+  };
+
+  start = () => {
+    this.setState({
+      isListVisible: true,
+    });
+  };
+
+  render() {
+    const { isListVisible, goodsList } = this.state;
+
+    if (isListVisible) {
+      return (
+        <GoodsList {...goodsList} />
+      );
+    }
```

Comment:
use conditional rendering inside jsx (inside return statement)
also no need to add goodsList to App's state and pass goodsList to GoodsList component
```suggestion
    if (isListVisible) {
      return (
        <GoodsList />
      );
    }
```